ListingService/Sources/ListingsDataSource.swift

```diff
@@ -25,14 +26,28 @@ public class ListingsDataSource {
 
         self.loadMoreTask?.cancel()
         self.loadMoreTask = nil
-        self.refreshTask = CoreAPI.popularListings { [weak self] result in
-            self?.refreshTask = nil
-            switch result {
-            case .success(let listings):
-                self?.listings = listings
-                self?.updated?()
-            case .error:
-                return
+        switch listingFeed {
```

Comment:
We will need to make this more scalable for many subreddits in the future. You can merge it in and I will refactor CoreAPI to have a single method for fetching any subreddit.

Reply:
Agreed!